File: ml/src/main/scala/ai/h2o/sparkling/ml/params/HasInputCols.scala
package ai.h2o.sparkling.ml.params

trait HasInputCols extends HasInputColsOnMOJO {
  def setInputCols(columns: Array[String]): this.type = set(inputCols -> columns)

  def setInputCols(column: String, otherColumns: String*): this.type = setInputCols(Array(column) ++ otherColumns)
}

File: py-scoring/src/ai/h2o/sparkling/ml/params/HasInputCols.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
from pyspark.ml.param import *


class HasInputCols(Params):
    inputCols = Param(
        Params._dummy(),
        "inputCols",
        "Names of input columns",
        H2OTypeConverters.toNullableListString())

    def getInputCols(self):
        return self.getOrDefault(self.inputCols)

    def setInputCols(self, value):
        return self._set(inputCols=value)
